Samples/Widgets/MultipleWidgets/cpp-win32-packaged/SampleWidgetProviderApp/WidgetProvider.cpp

```diff
@@ -0,0 +1,175 @@
+}
+
+// Handle the DeleteWidget call. This is notifying you that
+// you don't need to provide new content for the given WidgetId
+// since the user has unpinned the widget or it was deleted by the Host
+// for any other reason.
+void WidgetProvider::DeleteWidget(winrt::hstring const& widgetId, [[maybe_unused]] winrt::hstring const& customState)
+{
+	m_runningWidgetImpl.erase(widgetId);
+}
+
+// Handle the OnActionInvoked call. This function call is fired when the user's
+// interaction with the widget resulted in an execution of one of the defined
+// actions that you've indicated in the template of the Widget.
+// For example: clicking a button or submitting input.
+void WidgetProvider::OnActionInvoked(winrt::WidgetActionInvokedArgs actionInvokedArgs)
+{
+	auto widgetId = actionInvokedArgs.WidgetContext().Id();
+	auto runningWidget = FindRunningWidget(widgetId);
+	if (runningWidget)
+	{
+		runningWidget->OnActionInvoked(actionInvokedArgs);
+	}
+}
+
+// Handle the WidgetContextChanged call. This function is called when the context a widget
+// has changed. Currently it only signals that the user has changed the size of the widget.
+// There are 2 ways to respond to this event:
+// 1) Call UpdateWidget() with the new data/template to fit the new requested size.
+// 2) If previously sent data/template accounts for various sizes based on $host.widgetSize - you can use this event solely for telemtry.
+void WidgetProvider::OnWidgetContextChanged(winrt::WidgetContextChangedArgs contextChangedArgs)
+{
+	auto widgetContext = contextChangedArgs.WidgetContext();
+	auto widgetId = widgetContext.Id();
+	auto runningWidget = FindRunningWidget(widgetId);
+	if (runningWidget)
+	{
+		runningWidget->OnWidgetContextChanged(contextChangedArgs);
+		// Optionally: if the data/template for the new context is different
+		// from the previously sent data/template - send an update.
+	}
+}
+
+// Handle the Activate call. This function is called when widgets host starts listening
+// to the widget updates. It generally happens when the widget becomes visible and the updates
+// will be promptly displayed to the user. It's recommended to start sending updates from this moment
+// until Deactivate function was called.
+void WidgetProvider::Activate(winrt::Microsoft::Windows::Widgets::Providers::WidgetContext widgetContext)
+{
+	auto widgetId = widgetContext.Id();
+	auto runningWidget = FindRunningWidget(widgetId);
+	if (runningWidget)
+	{
+		runningWidget->Activate(widgetContext);
+	}
+}
+
+// Handle the Deactivate call. This function is called when widgets host stops listening
+// to the widget updates. It generally happens when the widget is not visible to the user
+// anymore and any further updates won't be displayed until the widget is visible again.
+// It's recommended to stop sending updates until Activate function was called.
+void WidgetProvider::Deactivate(winrt::hstring widgetId)
+{
+	auto runningWidget = FindRunningWidget(widgetId);
```

Comment:
Style nit:

```c++
if (auto runningWidget = FindRunningWidget(widgetId))
{
    runningWidget->Deactivate(widgetId);
}
```

Applies to basically all patterns like:

```c++
T foo = GetFoo(bar);
if (foo)
{
    foo->Zot();
}
```

Reply:
fixed with style